sdks/python/pom.xml

```diff
@@ -39,131 +39,165 @@
+                <goals>
+                  <goal>exec</goal>
+                </goals>
+                <configuration>
+                  <executable>mkdir</executable>
+                  <arguments>
+                    <argument>-p</argument>
+                    <argument>${python.user.base}</argument>
+                  </arguments>
+                </configuration>
+              </execution>
+              <execution>
+                <id>setup-test-tox</id>
+                <phase>process-test-resources</phase>
+                <goals>
+                  <goal>exec</goal>
+                </goals>
+                <configuration>
+                  <executable>${python.pip.bin}</executable>
+                  <arguments>
+                    <argument>install</argument>
+                    <argument>--user</argument>
+                    <argument>--upgrade</argument>
+                    <argument>--ignore-installed</argument>
+                    <argument>tox</argument>
+                  </arguments>
+                  <environmentVariables>
+                    <PYTHONUSERBASE>${python.user.base}</PYTHONUSERBASE>
+                  </environmentVariables>
+                </configuration>
+              </execution>
+              <execution>
+                <id>setuptools-test</id>
+                <phase>test</phase>
+                <goals>
+                  <goal>exec</goal>
+                </goals>
+                <configuration>
+                  <executable>${python.user.base}/bin/tox</executable>
+                  <arguments>
+                    <argument>-e</argument>
+                    <argument>py27</argument>
+                    <argument>-c</argument>
+                    <argument>tox.ini</argument>
+                  </arguments>
+                  <environmentVariables>
+                    <PYTHONUSERBASE>${python.user.base}</PYTHONUSERBASE>
+                  </environmentVariables>
+                </configuration>
+              </execution>
+              <execution>
+                <id>setuptools-sdist</id>
+                <phase>package</phase>
+                <goals>
+                  <goal>exec</goal>
+                </goals>
+                <configuration>
+                  <executable>${python.interpreter.bin}</executable>
+                  <arguments>
+                    <argument>setup.py</argument>
+                    <argument>sdist</argument>
+                    <argument>--dist-dir</argument>
+                    <argument>${project.build.directory}</argument>
+                  </arguments>
+                  <environmentVariables>
+                    <PYTHONUSERBASE>${python.user.base}</PYTHONUSERBASE>
+                  </environmentVariables>
+                </configuration>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+    <profile>
+      <id>beam-sdks-python-pydoc</id>
+      <activation>
```

Comment:
We have a misunderstanding -- let's reset.

There are 2 profiles:
* The comment about the first profile is above.
* The comment about Pydoc one is here.

My understanding:
* Runs on Linux-like OS only, because it requires `bash`. (Separately, converting to a python script might be better, since it presumably wouldn't be platform specific.)
* Should be in a profile that is _not_ on by default, exactly as Javadoc is, so that build time is not negatively affected for development scenarios.
* The profile should be on for pre-commit coverage, i.e., if Pydoc stops building, pre-commit should fail.
* The profile should be automatically activated when running a release (by virtue of the release plugin and release profile).

Happy to help with any of this, as you see fit ;-)